e2e/windows/vpack.ps1

```diff
@@ -0,0 +1,80 @@
+param(
+    [string]$arg1, #GitBranch
+    [string]$arg2, #VPackName
+    [string]$arg3, #VPackFolder
+    [string]$arg4, #UserName
+    [string]$arg5  #PAT
+)
+
+function Retry-Command {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Position=0, Mandatory=$true)]
+        [scriptblock]$ScriptBlock,
+
+        [Parameter(Position=1, Mandatory=$true)]
+        [string]$ErrorMessage,
+
+        [Parameter(Position=2, Mandatory=$false)]
+        [int]$Maximum = 5,
+
+        [Parameter(Position=3, Mandatory=$false)]
+        [int]$Delay = 10
+    )
+
+    Begin {
+        $cnt = 0
+    }
+
+    Process {
+        do {
+            $cnt++
+            try {
+                $ScriptBlock.Invoke()
+                if ($LASTEXITCODE) {
+                    throw "Retry $cnt : $ErrorMessage"
+                }
+                return
+            } catch {
+                Write-Output $_.Exception.InnerException.Message
+                Write-Output "Retry $cnt : $ScriptBlock"
+                Start-Sleep $Delay
+            }
+        } while ($cnt -lt $Maximum)
+
+        # Throw an error after $Maximum unsuccessful invocations. Doesn't need
+        # a condition, since the function returns upon successful invocation.
+        throw "All retries failed. $ErrorMessage"
+    }
+}
+
+cd C:\
+
+# Pull AgentBaker Repo
+Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
```

Comment:
Download the three files directly with the given branch?

Reply:
Thanks for the advice. I have modified it now.